### PR TITLE
libfixmath: new port

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

New port: https://github.com/PetteriAimonen/libfixmath

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.6.8 Server
Xcode 3.2.6

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

All tests pass in Rosetta:
```
Running tests...
/opt/local/bin/ctest --force-new-ctest-process 
Test project /opt/local/var/macports/build/_opt_PPCRosettaPorts_math_libfixmath/libfixmath/work/build
      Start  1: tests_ro64
 1/12 Test  #1: tests_ro64 .......................   Passed    0.02 sec
      Start  2: tests_no64
 2/12 Test  #2: tests_no64 .......................   Passed    0.02 sec
      Start  3: tests_rn64
 3/12 Test  #3: tests_rn64 .......................   Passed    0.02 sec
      Start  4: tests_nn64
 4/12 Test  #4: tests_nn64 .......................   Passed    0.02 sec
      Start  5: tests_ro32
 5/12 Test  #5: tests_ro32 .......................   Passed    0.02 sec
      Start  6: tests_no32
 6/12 Test  #6: tests_no32 .......................   Passed    0.02 sec
      Start  7: tests_rn32
 7/12 Test  #7: tests_rn32 .......................   Passed    0.02 sec
      Start  8: tests_nn32
 8/12 Test  #8: tests_nn32 .......................   Passed    0.02 sec
      Start  9: tests_ro08
 9/12 Test  #9: tests_ro08 .......................   Passed    0.03 sec
      Start 10: tests_no08
10/12 Test #10: tests_no08 .......................   Passed    0.02 sec
      Start 11: tests_rn08
11/12 Test #11: tests_rn08 .......................   Passed    0.02 sec
      Start 12: tests_nn08
12/12 Test #12: tests_nn08 .......................   Passed    0.02 sec

100% tests passed, 0 tests failed out of 12
```

### CODE REVIEW

File: math/libfixmath/Portfile [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+PortGroup           github 1.0
+
+github.setup        PetteriAimonen libfixmath ddff910fc03c6e9581ed186268b508dc0f51f4b7
+version             2022.12.27
+revision            0
+categories          math
+license             MIT
+maintainers         {@barracuda156 gmail.com:vital.had} openmaintainer
+description         Q16.16 format fixed point operations in C
+long_description    ${description}
```
Comment (herbygillot) [inline math/libfixmath/Portfile:14]: When reusing `${description}` in `long_description`, you need to use the Tcl expansion operator (`{*}`):

```suggestion
long_description    {*}${description}
```

Comment (barracuda156) [reply math/libfixmath/Portfile:14]: @herbygillot Sure, gimme a min.

**Comment by barracuda156** [comment]
@herbygillot Thank you!

**Comment by barracuda156** [comment]
Apparently old Intel systems also do not support `libubsan`?
Build succeeds on everything from Sierra up and fails on everything from El Captain down: https://ports.macports.org/port/libfixmath/details/
(PPC builds fine, but I have disabled `libubsan` linking there.)

```
ld: file not found: /opt/local/libexec/llvm-11/lib/clang/11.1.0/lib/darwin/libclang_rt.ubsan_osx_dynamic.dylib
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
Well, I can patch this out then, so that everything builds.

**Comment by barracuda156** [comment]
@herbygillot Will this work for old Intel systems? https://github.com/barracuda156/macports-ports/commit/265cd5560208988166487f373831d60b475e2de9

**Pull request merged**